crates/extensions/tedge_config_manager/src/tests.rs

```diff
@@ -0,0 +1,439 @@
+            ).with_retain())
+        );
+
+    // This message being published over MQTT is also received by the config-manager itself
+    mqtt.send(executing_message.unwrap()).await?;
+
+    // Then uploads the requested content over HTTP
+    let actual_request = http.recv().await;
+    let expected_request = Some(
+        HttpRequestBuilder::put(
+            "http://127.0.0.1:3000/tedge/file-transfer/main/config-snapshot/type_two-1234",
+        )
+        .header("Content-Type", "text/plain")
+        .body("filename: file_b\nSome content\n".to_string())
+        .build()
+        .unwrap(),
+    );
+    assert_request_eq(actual_request, expected_request);
+
+    // File transfer responds with 201
+    let response = HttpResponseBuilder::new().status(201).build().unwrap();
+    http.send(Ok(response)).await?;
+
+    // Finally, the config manager notifies that request was successfully processed
+    assert_eq!(
+            mqtt.recv().await,
+            Some(MqttMessage::new(
+                &config_topic,
+                format!(r#"{{"status":"successful","tedgeUrl":"http://127.0.0.1:3000/tedge/file-transfer/main/config-snapshot/type_two-1234","type":"type_two","path":{:?}}}"#, tempdir.path().join("file_b"))
+            ).with_retain())
+        );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn config_manager_download_update() -> Result<(), anyhow::Error> {
+    let tempdir = prepare()?;
+    let (mut mqtt, _http, _fs, mut downloader) = spawn_config_manager_actor(tempdir.path());
+
+    let config_topic = Topic::new_unchecked("te/device/main///cmd/config_update/1234");
+
+    // Let's ignore the reload messages sent on start
+    mqtt.skip(2).await;
+
+    // When a config snapshot request is received
+    let snapshot_request = r#"
+        {
+            "status": "init",
+            "tedgeUrl": "http://127.0.0.1:3000/tedge/file-transfer/main/config_update/type_two-1234",
+            "remoteUrl": "http://www.remote.url",
+            "type": "type_two"
+        }"#;
+
+    mqtt.send(MqttMessage::new(&config_topic, snapshot_request).with_retain())
+        .await?;
+
+    // The config manager notifies that the request has been received and is processed
+    let executing_message = mqtt.recv().await;
+    assert_eq!(
+        executing_message,
+            Some(MqttMessage::new(
+                &config_topic,
+                r#"{"status":"executing","tedgeUrl":"http://127.0.0.1:3000/tedge/file-transfer/main/config_update/type_two-1234","remoteUrl":"http://www.remote.url","type":"type_two"}"#
+            ).with_retain())
+        );
+
+    // This message being published over MQTT is also received by the config-manager itself
+    mqtt.send(executing_message.unwrap()).await?;
+
+    // Assert config download request.
+    let (id, download_request) = downloader.recv().await.unwrap();
```

Comment:
I guess this `id` is tedgeUrl. There is value to add an assert for that since this message is created by config-manager.

Reply:
Not exactly, `id` represents `topic` that is used to send mqtt message after successful download. However, it is probably good idea to rename it and check if that topic is correct.